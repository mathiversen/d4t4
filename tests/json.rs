use data_parser::{Data, Result};
use indoc::indoc;
use insta::assert_json_snapshot;

// https://github.com/pest-parser/pest/blob/master/grammars/tests/examples.json
#[test]
fn it_can_parse_json() -> Result<()> {
    let markup = indoc!(
        r#"  {
            "null": null,
            "true": true,
            "false": false,
            "":  23456789012E66,
            "e": 0.123456789e-12,
            "E": 1.234567890E+34,
            "integer": 1234567890,
            "one": 1,
            "real": -9876.543210,
            "zero": 0,
        }"#
    );
    let x = Data::parse(markup)?;
    assert_json_snapshot!(x);
    Ok(())
}
