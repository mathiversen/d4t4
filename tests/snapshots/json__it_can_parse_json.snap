---
source: tests/json.rs
expression: x
---
{
  "": 2.3456789012e76,
  "E": 1.23456789e34,
  "e": 1.23456789e-13,
  "false": false,
  "integer": 1234567890.0,
  "null": null,
  "one": 1.0,
  "real": -9876.54321,
  "true": true,
  "zero": 0.0
}
